File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        long secondHalf=0,firstHalf=0;
        int res=0;

        for(int i:nums){
            secondHalf+=i;
        }
      

        for(int i=0;i<nums.length-1;i++){
            secondHalf-=nums[i];
            firstHalf+=nums[i];

            if (firstHalf >= secondHalf){
                res++;
            }
        }
return res;
        
    }
}

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:

        second_half=sum(nums)
        first_half=0
        res=0
        for i in range(len(nums)-1):
            second_half-=nums[i]
            first_half+=nums[i]
            if first_half >= second_half:
                res+=1
        return res
           

      
        